config/platform.go

```diff
@@ -0,0 +1,108 @@
+package config
+
+import (
+	"bufio"
+	"context"
+	"os"
+	"os/exec"
+	"regexp"
+	"runtime"
+	"strings"
+	"time"
+
+	"go.viam.com/rdk/logging"
+)
+
+var (
+	cudaRegex            = regexp.MustCompile(`Cuda compilation tools, release (\d+)\.`)
+	aptCacheVersionRegex = regexp.MustCompile(`\nVersion: (\d+)\D`)
+	savedPlatformTags    []string
+)
+
+// helper to read platform tags for GPU-related system libraries.
+func readGPUTags(tags []string) []string {
+	// this timeout is for all steps in this function.
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
```

Comment:
on a random device the commands take:
- `nvcc --version` 10ms
- `apt-cache show` 140ms

added the timeout because I didn't want to block forever, but have not observed these commands to be slow

(one unknown that I will check -- will `apt-cache show` hang if another process has the apt lock)

open to narrowing the timeout if you want

getting this wrong means downloading the wrong build; may be worth thinking about how to make this robust without blocking startup

Reply:
if we could narrow the timeout a bit it would be nice (3-5s seems like it may give enough of a leeway?), just so we have a good idea of the upper bound

Reply:
apt-cache does not need the apt lock file. with an apt command running in the background, apt update in foreground fails with lock error, apt-cache succeeds

```sh
$ sudo apt update
Reading package lists... Done
E: Could not get lock /var/lib/apt/lists/lock. It is held by process 1479846 (apt)
N: Be aware that removing the lock file is not a solution and may break your system.
E: Unable to lock directory /var/lib/apt/lists/

$ time apt-cache show libc6
Package: libc6
Architecture: amd64
...

$ sudo apt update
Reading package lists... Done
E: Could not get lock /var/lib/apt/lists/lock. It is held by process 1479846 (apt)
N: Be aware that removing the lock file is not a solution and may break your system.
E: Unable to lock directory /var/lib/apt/lists/
```

Reply:
So if these functions should never block, a timeout of 1 second would be overkill, yeah?